test/fast/Unit tests/nvm_print_versions

```diff
@@ -0,0 +1,139 @@
+#!/bin/sh
+
+# shellcheck disable=SC2317
+
+die () { echo "$@" ; cleanup ; exit 1; }
+
+cleanup() {
+  unset -f nvm_remote_versions nvm_ls nvm_ls_current nvm_remote_versions
+  if [ -n "$TEMP_NVM_MIN" ]; then
+    export NVM_MIN="$TEMP_NVM_MIN"
+  fi
+}
+
+\. ../../../nvm.sh
+
+
+if [ -n "$NVM_MIN" ]; then
+  TEMP_NVM_MIN="$NVM_MIN"
+  unset NVM_MIN
+fi
+
+# mock currently installed versions
+nvm_ls() {
+  echo "v16.20.2
+v18.20.3
+system"
+}
+
+# mock currently active version
+nvm_ls_current() {
+  echo "v18.20.3"
+}
+
+nvm_remote_versions() {
+  echo "v16.0.0
+v16.20.2 Gallium *
+v17.0.0
+v17.9.1
+v18.0.0
+v18.1.0
+v18.20.2 Hydrogen
+v18.20.3 Hydrogen *
+v19.0.0
+v19.9.0
+v20.0.0
+v20.8.1
+v20.9.0 Iron *
+v21.0.0
+v21.1.0"
+}
+
+
+# nvm_print_versions should print all versions from nvm_remote_versions
+OUTPUT="$(NVM_NO_COLORS=1 nvm_print_versions "$(nvm_remote_versions)" | sed -r 's/^[ \t]+//')"
+EXPECTED_OUTPUT="v16.0.0
+v16.20.2 * (Latest LTS: Gallium)
+v17.0.0
+v17.9.1
+v18.0.0
+v18.1.0
+v18.20.2   (LTS: Hydrogen)
+->     v18.20.3 * (Latest LTS: Hydrogen)
+v19.0.0
+v19.9.0
+v20.0.0
+v20.8.1
+v20.9.0   (Latest LTS: Iron)
+v21.0.0
+v21.1.0"
+
+[ "_$OUTPUT" = "_$EXPECTED_OUTPUT" ] || die "nvm_print_versions did not output all expected versions; got $OUTPUT"
+
+
+# versions lower than 18 should be filtered out, but v16.20.2 should be kept since it's installed
+OUTPUT="$(NVM_NO_COLORS=1 NVM_MIN=18 nvm_print_versions "$(nvm_remote_versions)" | sed -r 's/^[ \t]+//')"
```

Comment:
what if you have v16.20.1 installed, for example, and you still want to see that v16.20.2 is both available, and the latest LTS from that line?

maybe anything that's installed _plus_ `^` of it should still be shown?

or maybe, we should skip the "keep if installed" logic entirely?

Reply:
I think the chosen behavior is slightly more preferable, due to the fact that `nvm ls-remote` has always been highlighting the installed versions, so that people can intuitively compare the versions they installed, to the newer versions available. And that kind of intuition is needed for people to decide whether or when to install a new version, which is much nicer than having to run both `nvm ls-remote` and `nvm ls`.

Reply:
ok - in that case, setting a MIN_VER of 18, with 16.20.1 installed as the only 16, should still show 16.20.2 as an installable version.

Reply:
Hmmm, isn't that a bit too far? If we do that then it means the min version is effectively the oldest installed version, hence defeating the point of specifying a min version. Nevertheless I agree it's a bit tricky to decide on which is the sweet spot.

Another option could be that we first let it out and see the user perception from the community.

Reply:
In my scenario, 17 still wouldn’t be displayed. So, it’d display all installed versions, anything that can be safely upgraded to, as well as the min version and above.

Unfortunately community perception and feedback is in practice almost never sufficient enough to give an accurate picture of the actual community reception - just the most vocal members - so it’s important the project is happy with the way it works initially.

Reply:
Just clarify, for an installed version, with both newer patch versions and newer minor versions, do you expect to see just the patch version updates? or both?

Reply:
I’d expect both - everything up to and not including the next major

Reply:
Done, see https://github.com/nvm-sh/nvm/pull/3277/commits/477d22917

Now when we encounter a version that's installed, we make sure to not skip all following versions having the same major version.